src/sentry/static/sentry/app/components/dropdownAutoComplete.jsx

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import _ from 'lodash';
+import AutoComplete from './autoComplete';
+import Input from '../views/settings/components/forms/controls/input';
+
+class DropdownAutoComplete extends React.Component {
+  static propTypes = {
+    items: PropTypes.array,
+    isOpen: PropTypes.bool,
+    onSelect: PropTypes.func,
+    children: PropTypes.func,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+      selectedItem: undefined,
+    };
+  }
+
+  toggleMenu = () => this.setState({isOpen: !this.state.isOpen});
+
+  openMenu = e => this.setState({isOpen: true});
+
+  onSelect = selectedItem => {
+    this.setState({selectedItem});
+    if (this.props.onSelect) this.props.onSelect(selectedItem);
+    this.toggleMenu();
+  };
+
+  filterItems = (items, inputValue) =>
+    items.filter(item => {
+      return item.value.toLowerCase().indexOf(inputValue.toLowerCase()) > -1;
+    });
+
+  filterGroupedItems = (groups, inputValue) =>
+    groups
+      .map(group => {
+        return {
+          ...group,
+          items: this.filterItems(group.items, inputValue),
+        };
+      })
+      .filter(group => group.items.length > 0);
+
+  autoCompleteFilter = (items, inputValue) => {
+    let itemCount = 0;
+
+    return items[0].items
+      ? _.flatMap(this.filterGroupedItems(items, inputValue), item => {
+          return [
+            {...item.group},
+            ...item.items.map(groupedItem => ({...groupedItem, index: itemCount++})),
+          ];
+        })
+      : this.filterItems(items).map((item, index) => ({...item, index}));
+  };
+
+  render() {
+    return (
+      <div style={{position: 'relative', display: 'inline-block'}}>
+        {this.state.isOpen && (
+          <StyledMenu>
+            <AutoComplete itemToString={item => item.searchKey} onSelect={this.onSelect}>
+              {({
+                getRootProps,
+                getInputProps,
+                getMenuProps,
+                getItemProps,
+                inputValue,
+                selectedItem,
+                highlightedIndex,
+                isOpen,
+              }) => {
+                return (
+                  <div {...getRootProps()}>
+                    <StyledInputContainer>
+                      <StyledInput
+                        autoFocus
+                        {...getInputProps({onBlur: this.toggleMenu})}
+                      />
+                    </StyledInputContainer>
+                    <div {...getMenuProps()}>
+                      <div>
+                        {this.autoCompleteFilter(this.props.items, inputValue).map(
+                          (item, index) =>
+                            item.index || item.index == 0 ? (
```

Comment:
is the presence of `index` being used to decide if something is a group here? there's a `group` property on the groups that would be more self-explanatory, right?

Reply:
if you keep item.idex, you should check if it's defined (we have a `defined` fn in `utils`, and lodash has `_.isUndefined`)

Reply:
i killed it for simplicity, but in retrospect that seems like a confusing thing to do. I'm gonna go back and add the group property.